csrc/multidevice/executor.h

```diff
@@ -139,6 +135,12 @@ class MultiDeviceExecutor {
   std::unordered_map<SegmentedGroup*, bool> should_run_;
   // Cache whether a SegmentedGroup requires inter-device communication
   std::unordered_map<SegmentedGroup*, bool> is_resharding_;
+  // Cached objects used for MultiDevice allocation
+  std::unique_ptr<Fusion> allocator_fusion_;
+  // Cache the tensors that need to be allocated at runtime, which correspond to
+  // the destination buffers of interdevice communications.
+  std::vector<Val*> vals_to_allocate_;
```

Comment:
With this PR, we'll have both `val_to_IValue_` and `vals_to_allocate`. Is that a necessary complexity? According to https://github.com/NVIDIA/Fuser/pull/1900/files#diff-d783a79fdc189059065a81ba9acb6d6c3d3ecc5c70188706febc97d7fea5aa36R199-R201, the former seems to subsume the latter.

Reply:
I think it is not possible, at least in the present state. The problem would come from using multiple time the executor.
After being used at least once, `val_to_IValue_` will contain more keys than the one allocated, and so the second call to  https://github.com/NVIDIA/Fuser/pull/1900/files#diff-d783a79fdc189059065a81ba9acb6d6c3d3ecc5c70188706febc97d7fea5aa36R199-R201 will be wrong